lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,1115 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.9
+part of dart.ui;
+
+/// Callback type for [PlatformDispatcher.onPlatformConfigurationChanged].
+///
+/// Receives the new `configuration`.
+typedef PlatformConfigurationChangedCallback = void Function(PlatformConfiguration configuration);
+
+/// Platform event dispatcher singleton.
+///
+/// The most basic interface to the host operating system's interface.
+///
+/// This is the central entry point for platform messages and configuration
+/// events from the platform.
+///
+/// It exposes the core scheduler API, the input event callback, the graphics
+/// drawing API, and other such core services.
+///
+/// It manages the list of the application's [views] and the [screens] attached
+/// to the device, as well as the [configuration] of various platform
+/// attributes.
+///
+/// It is the type of the `WidgetsBinding.instance.platformDispatcher`
+/// singleton.
+///
+/// While there is also a [PlatformDispatcher.instance] singleton object in
+/// `dart:ui`, unless you are implementing your own binding, consider avoiding a
+/// static reference to it. See the documentation for
+/// [PlatformDispatcher.instance] for more details about why you might want to
+/// avoid using it directly.
+class PlatformDispatcher {
+  /// Private constructor, since only dart:ui is supposed to create one of
+  /// these.
+  PlatformDispatcher._() {
+    _setNeedsReportTimings = _nativeSetNeedsReportTimings;
+  }
+
+  /// The [PlatformDispatcher] singleton.
+  ///
+  /// Instead of accessing this directly, consider avoiding a static reference
+  /// to this and instead use a binding for dependency resolution such as
+  /// `WidgetsBinding.instance.platformDispatcher`.
+  ///
+  /// Static access of this object means that Flutter has few, if any options to
+  /// fake or mock the given object in tests. Even in cases where Dart offers
+  /// special language constructs to forcefully shadow such properties, those
+  /// mechanisms would only be reasonable for tests and they would not be
+  /// reasonable for a future of Flutter where we legitimately want to select an
+  /// appropriate implementation at runtime.
+  ///
+  /// Flutter's binding uses this instance, and if you are implementing your own
+  /// binding, of course, it may make sense to use the
+  /// [PlatformDispatcher.instance] object directly.
+  ///
+  /// If you need to access these APIs before invoking `runApp()`, but are using
+  /// Flutter bindings, instantiate the bindings yourself by calling
+  /// `WidgetsFlutterBinding.ensureInitialized()` and then use
+  /// `WidgetsBinding.instance.platformDispatcher`.
+  static final PlatformDispatcher instance = PlatformDispatcher._();
+
+  /// The current platform configuration.
+  ///
+  /// If values in this configuration change, [onMetricsChanged] will be called.
```

Comment:
Will onMetricsChanged be called in addition to onPlatformConfigurationChanged? Or is this a typo?